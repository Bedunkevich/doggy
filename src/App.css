.App-main__leftbar {
  position: absolute;
  left: 0px;
  top: 0px;
  bottom: 0px;
  width: 200px;
  overflow-x: hidden;
  overflow-y: auto;
}

.App-main__image-galery {
  position: absolute;
  left: 200px;
  top: 0px;
  bottom: 0px;
  right: 0px;
  border-left: 1px solid #ccc;
  overflow: hidden;
}

.App-main__image-galery > div {
  font-size: 11px;
  padding: 20px;
  line-height: 200%;
  width: 50%;
}

input.App-main__search  {
  font-family: 'Roboto', -apple-system, Arial;
  margin: 10px;
  border-radius: 15px;
  padding: 5px 10px;
  border: 1px solid rgba(0, 0, 0, .2);
  outline: none;
  width: 160px;
}

.App-main__search-input {
  background-color: rgba(255, 255, 255, 0.9);
  position: sticky;
  top: 0px;
}

.App-main__breed-list {
  margin-bottom: 20px;
}

.App-main__breed-list > ul,
.App-main__breed-list > ul > li > ul {
  padding: 0px;
  margin: 0px;
  margin-left: 10px;
}

.App-main__breed-list > ul > li > ul > li {
  list-style: square;
  margin: 0px;
  margin-left: 20px;
}

