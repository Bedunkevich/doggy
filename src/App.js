// eslint-disable-next-line
import styles from 'react-responsive-carousel/lib/styles/carousel.min.css';
import React, { Component } from 'react';
import './App.css';
import { getAllDogs } from './api';
import DogsList from './components/DogsList';
import PicturesGallery from './components/PicturesGallery';


class App extends Component {

  constructor(props) {
    super(props);
    this.state = { dogsData: null, selectedDog: null };

    getAllDogs().then(data => {
      const {message} = data.response;
      this.setState({ dogsData: message });
    }, error => {
      console.log(error);
    });
  }

  loadPictures = (dog) => {
    this.setState({ selectedDog: dog });
  }

  render() {
    return (
      <div className="App-main">
        <div className='App-main__leftbar'>
          <DogsList loadPictures={this.loadPictures} dogsData={this.state.dogsData} />
        </div>
        <div className="App-main__image-galery">
          <PicturesGallery selectedDog={this.state.selectedDog} />
        </div>
      </div>
    );
  }
}

export default App;
