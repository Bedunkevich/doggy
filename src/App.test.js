import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { mount, render } from 'enzyme';

describe('Doggy tests', () => {
  fetch.mockResponse(JSON.stringify({status: 'success', message: { breed : ['sub_breed'] }}), { status: 200 });

  it('Expect App toMatchSnapshot', () => {
    const app = render(<App />);
    expect.assertions(1);
    expect(app).toMatchSnapshot();
  });

});


