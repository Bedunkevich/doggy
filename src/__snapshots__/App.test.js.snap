// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doggy tests Expect App toMatchSnapshot 1`] = `
<div
  class="App-main"
>
  <div
    class="App-main__leftbar"
  >
    <div>
      Loading...
    </div>
  </div>
  <div
    class="App-main__image-galery"
  >
    <div>
      Select the breed
    </div>
  </div>
</div>
`;
